src/test/java/org/folio/rest/support/kafka/FakeKafkaConsumer.java

```diff
@@ -127,13 +128,35 @@ public static Collection<KafkaConsumerRecord<String, JsonObject> > getHoldingsEv
   private static KafkaConsumerRecord<String, JsonObject>  getLastEvent(
     Collection<KafkaConsumerRecord<String, JsonObject> > events) {
 
-    return events.stream().skip(events.size() - 1).findFirst().orElse(null);
+    Iterator<KafkaConsumerRecord<String, JsonObject>> iter = events.stream().iterator();
+    KafkaConsumerRecord<String, JsonObject> last = null;
+
+    while (iter.hasNext()) {
+      KafkaConsumerRecord<String, JsonObject> record = iter.next();
+
+      if (last == null || record.timestamp() > last.timestamp()) {
+        last = record;
+      }
+    }
+
+    return last;
```

Comment:
I agree that imposing a defined order on this is a reasonable activity (though it is a somewhat false assumption that messages are published in timestamp order).

I would prefer a more intent revealing implementation if possible, especially given this is test code that won't have it's own tests. 

What led to the use of an iterator instead of something like `stream.sorted` or `collections.sort`?

Reply:
> What led to the use of an iterator instead of something like stream.sorted or collections.sort?
I wasn't focusing as much on the details as I probably should have.
The existing code was using events.stream() and I just continued using it without thinking much about it as my focus was more on the test failures.

I'm all for improving this and I also am concerned about the assumption in timestamp order as well.

Reply:
I suggest adding a comment making the assumption about the timestamp order being reliable.

And experiment with using a more expressive way of achieving the sorted list.